Cue/app/tabs/MenuProfileItem.js

```diff
@@ -36,20 +36,42 @@ const styles = {
 class MenuProfileItem extends React.Component {
   props: {
     name: string,
+    navigator: Navigator,
     extraStyles?: Object,
-    onLogOut: () => void,
 
     // From Redux:
     logOut: () => void,
+    localChanges: {},
   }
 
   _onMenuAction = (index: number) => {
-    this._logOut()
-  }
-
-  _logOut = () => {
-    LoginManager.logOut()
-    this.props.logOut()
+    if (this.props.localChanges && this.props.localChanges.length) {
+      this.props.syncLibrary(this.props.localChanges).then(failedSyncs =>{
+        if (failedSyncs && failedSyncs.length) {
+          Alert.alert(
+            'Failed to sync changes',
+            'Resolve conflicts or logout and lose local changes',
+            [
+              {text: "Logout", onPress: () => this.props.logOut()},
+              {text: "Resolve", onPress: () => this.props.navigator.push({failedSyncs})}
+            ],
```

Comment:
Also, is our logout action going to be called `Log Out` or `Sign Out`? Right now we're using `Sign Out` as the user-facing word, but since internally we're calling everything `logOut`, it might be better to be consistent and change the wording in the user-facing language as well (in `AccountHome` and `MenuProfileItem`).